modules/sso.php

```diff
@@ -674,7 +675,11 @@ function handle_login() {
 					return;
 				}
 
-				$user_found_with = Jetpack_SSO_Helpers::new_user_override()
+				if ( $new_user_override_role ) {
```

Comment:
Can we add tests that the role is getting set as expected?

My initial thoughts for this were that we would pass `$new_user_override_role` into `Jetpack_SSO_Helpers::generate_user()`, which would make it easier to test.